File: app/src/main/java/com/semisonfire/cloudgallery/ui/main/disk/DiskContract.java
package com.semisonfire.cloudgallery.ui.main.disk;

import com.semisonfire.cloudgallery.data.model.Photo;
import com.semisonfire.cloudgallery.ui.base.MvpPresenter;
import com.semisonfire.cloudgallery.ui.base.MvpView;

import java.util.List;

public interface DiskContract {

    interface View extends MvpView {

        void onPhotosLoaded(List<Photo> photos);

        void onUploadingPhotos(List<Photo> photos);

        void onPhotoUploaded(Photo photo);

        void onPhotoDownloaded(String path);

        void onPhotoDeleted(Photo photo);
    }

    interface Presenter<V extends View> extends MvpPresenter<V> {

        void getPhotos(int offset);

        void uploadPhotos(List<Photo> photos);

        void uploadPhoto(Photo photo);

    }

}

File: app/src/main/java/com/semisonfire/cloudgallery/ui/disk/DiskContract.kt
package com.semisonfire.cloudgallery.ui.disk

import com.semisonfire.cloudgallery.data.model.Photo
import com.semisonfire.cloudgallery.core.mvp.MvpPresenter
import com.semisonfire.cloudgallery.core.mvp.MvpView

interface DiskContract {

  interface View : MvpView {

    fun onPhotosLoaded(photos: List<Photo>)

    fun onUploadingPhotos(photos: List<Photo>)

    fun onPhotoUploaded(photo: Photo)

    fun onPhotoDownloaded(path: String)

    fun onPhotoDeleted(photo: Photo)
  }

  interface Presenter : MvpPresenter<View> {

    fun getPhotos(offset: Int)

    fun uploadPhotos(photos: List<Photo>)

    fun uploadPhoto(photo: Photo)
    fun getUploadingPhotos()
    fun downloadPhotos(photos: List<Photo>)
    fun deletePhotos(photos: List<Photo>)

  }

}
